src/QueryMiddleware.php

```diff
@@ -82,35 +84,83 @@ public function __construct(
 
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        $queryName = $request->getAttribute(self::NAME_ATTRIBUTE);
+        $messages = $this->parseRequestMessages($request);
 
-        if (null === $queryName) {
-            throw new RuntimeException(
-                sprintf('Query name attribute ("%s") was not found in request.', self::NAME_ATTRIBUTE),
-                StatusCodeInterface::STATUS_BAD_REQUEST
+        $responses = [];
+
+        foreach ($messages as $message) {
+            $message['metadata'] = $this->metadataGatherer->getFromRequest($request);
+
+            $query = $this->queryFactory->createMessageFromArray(
+                $message[self::NAME_ATTRIBUTE],
+                $message
             );
-        }
-        $payload = $request->getQueryParams();
 
-        if ($request->getMethod() === RequestMethodInterface::METHOD_POST) {
-            $payload['data'] = $request->getParsedBody();
+            try {
+                $responses[] = $this->queryBus->dispatch($query);
```

Comment:
This array contains "promises", not responses. Maybe we should name it correctly.
Also this array should get named keys, since the `promise::all()` method would than return an array with same keys for each response. Than it would be much easier to identify the result for each query in the response.

Reply:
Good idea to change the name of the variable.